### ISSUE TITLE
Typescript issue with where function

### ISSUE THREAD

**Issue by samalexander** [sequelize/sequelize#10724]
## What are you doing?

When calling `sequelize.where` and passing a condition that contains an operator symbol, a typescript error occurs. For example:
```js
where('createdAt', {
    [Op.lt]: new Date()
})
```
This code was previously working when using `@types/sequelize` for typescript definitions. The move to built-in type `LogicType` for the logic parameter has caused the issue.

I have created a minimal reproduction [here](https://github.com/samalexander/sequelize-where).

## What do you expect to happen?
To accept the use of an operator symbol when building a where condition.

## What is actually happening?
A typescript error:

```
Argument of type '{ [Op.lt]: Date; }' is not assignable to parameter of type 'LogicType'.
  Object literal may only specify known properties, and '[Op.lt]' does not exist in type 'AndOperator | OrOperator | Fn | Col | Literal'.
```

__Dialect:__ any
__Dialect version:__ n/a
__Database version:__ n/a
__Sequelize version:__ 5.2.12
__Tested with latest release:__ Yes - as above

**Comment by SimonSchick** [comment]
@sushantdhiman can you clarify on the nature of `where` and/or `Utils.Where`? Documentation for it is a little vague and partially misleading (refering to `Symbol`s as default values while actually using `=` etc.).

**Comment by IMalaniak** [comment]
The same error is in 6.x now

**Comment by tanmayaBiswalOdiware** [comment]
Dude this error still exists and typescript is crying about it. 😖

**Comment by ephys** [comment]
The typings surrounding that utility were completely rewritten since 6.3. The issue may be fixed in v6.35.2+ and/or v7.0.0-alpha.34+

**Comment by tanmayaBiswalOdiware** [comment]
It seems to have fixed itself for now. Was on 6.32.1 version. Still on same version but works now. So ***shrug***
